codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2HeadersDecoderTest.java

```diff
@@ -129,9 +133,50 @@ public void execute() throws Throwable {
 
     @Test
     public void duplicatePseudoHeadersMustFailValidation() throws Exception {
-        final ByteBuf buf = encode(b(":authority"), b("abc"), b(":authority"), b("def"));
+        final DefaultHttp2HeadersDecoder decoder = new DefaultHttp2HeadersDecoder(true);
+        verifyValidationFails(decoder, encode(b(":authority"), b("abc"), b(":authority"), b("def")));
+    }
+
+    @Test
+    void decodingTrailersTeHeaderMustNotFailValidation() throws Exception {
```

Comment:
Would be nice to have a test that validates the remote peer receives a reset frame with `PROTOCOL_ERROR`

Reply:
Looks like we don't have any integration tests.

@normanmaurer Do you know of a good place to add a test like this, or is this an entirely new test?